picoquictest/stresstest.c

```diff
@@ -0,0 +1,875 @@
+    picoquic_call_back_event_t fin_or_event, void* callback_ctx)
+{
+    int ret = 0;
+    picoquic_stress_server_callback_ctx_t* ctx = (picoquic_stress_server_callback_ctx_t*)callback_ctx;
+
+    if (fin_or_event == picoquic_callback_close || fin_or_event == picoquic_callback_application_close) {
+        if (ctx != NULL) {
+            free(ctx);
+            picoquic_set_callback(cnx, stress_server_callback, NULL);
+        }
+    }
+    else if (fin_or_event == picoquic_callback_challenge_response) {
+        /* Do nothing */
+    }
+    else {
+        if (ctx == NULL) {
+            picoquic_stress_server_callback_ctx_t* new_ctx = (picoquic_stress_server_callback_ctx_t*)
+                malloc(sizeof(picoquic_stress_server_callback_ctx_t));
+            if (new_ctx == NULL) {
+                /* Should really be a debug-break error */
+                picoquic_close(cnx, PICOQUIC_ERROR_MEMORY);
+                stress_debug_break();
+            }
+            else {
+                memset(new_ctx, 0, sizeof(picoquic_stress_server_callback_ctx_t));
+                picoquic_set_callback(cnx, stress_server_callback, new_ctx);
+                ctx = new_ctx;
+            }
+        }
+
+        if (ctx != NULL) {
+            /* verify state and copy data to the stream buffer */
+            if (fin_or_event == picoquic_callback_stop_sending) {
+                if ((ret = picoquic_reset_stream(cnx, stream_id, 0)) != 0) {
+                    stress_debug_break();
+                }
+            }
+            else if (fin_or_event == picoquic_callback_stream_reset) {
+                if ((ret = picoquic_reset_stream(cnx, stream_id, 0)) != 0) {
+                    stress_debug_break();
+                }
+            }
+            else {
+                /* Write a response, which should somehow depend on the stream data and
+                * the stream status and the data bytes */
+                if ((stream_id & 3) != 0) {
+                    /* This is not a client-initiated bidir stream. Just ignore the data */
+                }
+                else {
+                    uint64_t bidir_id = stream_id / 4;
+                    size_t response_length = 0;
+
+
+                    if (bidir_id < PICOQUIC_STRESS_MAX_NUMBER_TRACKED_STREAMS) {
+                        size_t received = ctx->data_received_on_stream[bidir_id] + length;
+                        if (ctx->data_received_on_stream[bidir_id] < PICOQUIC_STRESS_MINIMAL_QUERY_SIZE) {
+                            int processed = (int) length;
+                            if (received >= PICOQUIC_STRESS_MINIMAL_QUERY_SIZE) {
+                                processed = (int) received - PICOQUIC_STRESS_MINIMAL_QUERY_SIZE;
+                            }
+
+                            for (int i = 0; i < processed; i++) {
+                                ctx->data_sum_of_stream[bidir_id] =
+                                    ctx->data_sum_of_stream[bidir_id] * 101 + bytes[i];
```

Comment:
I don't know what you are doing here, why this multiplication with `101`? Could you maybe add a comment what all this is doing?

Reply:
I did add a comment. Is that OK?